src/silo/query_engine/filter_expressions/has_mutation.cpp

```diff
@@ -22,63 +22,73 @@ class DatabasePartition;
       position(position) {}
 
-std::string HasMutation::toString(const silo::Database& /*database*/) const {
-   const std::string nuc_sequence_name_prefix =
-      nuc_sequence_name ? nuc_sequence_name.value() + ":" : "";
-   return nuc_sequence_name_prefix + std::to_string(position);
+template <typename SymbolType>
+std::string HasMutation<SymbolType>::toString(const silo::Database& /*database*/) const {
+   const std::string sequence_name_prefix = sequence_name ? sequence_name.value() + ":" : "";
+   return sequence_name_prefix + std::to_string(position);
 }
 
-std::unique_ptr<operators::Operator> HasMutation::compile(
+template <typename SymbolType>
+std::unique_ptr<operators::Operator> HasMutation<SymbolType>::compile(
    const silo::Database& database,
    const silo::DatabasePartition& database_partition,
    AmbiguityMode mode
 ) const {
-   const std::string nuc_sequence_name_or_default =
-      nuc_sequence_name.value_or(database.database_config.default_nucleotide_sequence);
    CHECK_SILO_QUERY(
-      database.nuc_sequences.contains(nuc_sequence_name_or_default),
-      "Database does not contain the nucleotide sequence with name: '" +
-         nuc_sequence_name_or_default + "'"
+      sequence_name.has_value() || database.getDefaultSequenceName<SymbolType>().has_value(),
+      fmt::format(
+         "Database does not have a default sequence name for {} Sequences", SymbolType::SYMBOL_NAME
+      )
+   );
+   const std::string sequence_name_or_default =
+      sequence_name.has_value() ? sequence_name.value()
+                                : database.getDefaultSequenceName<SymbolType>().value();
+   CHECK_SILO_QUERY(
+      database.getSequenceStores<SymbolType>().contains(sequence_name_or_default),
+      fmt::format(
+         "Database does not contain the {} sequence with name: '{}'",
+         SymbolType::SYMBOL_NAME,
+         sequence_name_or_default
+      )
    )
 
-   const Nucleotide::Symbol ref_symbol =
-      database.nuc_sequences.at(nuc_sequence_name_or_default).reference_sequence.at(position);
+   auto ref_symbol = database.getSequenceStores<SymbolType>()
+                        .at(sequence_name_or_default)
+                        .reference_sequence.at(position);
 
-   if (mode == UPPER_BOUND) {
-      auto expression = std::make_unique<Negation>(std::make_unique<NucleotideSymbolEquals>(
-         nuc_sequence_name_or_default, position, ref_symbol
-      ));
-      return expression->compile(database, database_partition, NONE);
+   std::vector<typename SymbolType::Symbol> symbols =
+      std::vector(SymbolType::SYMBOLS.begin(), SymbolType::SYMBOLS.end());
+   if (mode == AmbiguityMode::UPPER_BOUND) {
+      // We can only be sure, that the symbol did not mutate, if the ref_symbol is at that position
+      symbols.erase(std::remove(symbols.begin(), symbols.end(), ref_symbol), symbols.end());
+   } else {
+      // Remove all symbols that could match the searched base
```

Comment:
Do we need this comment?

Reply:
I like it, as it is not obvious, and the function name approach would be kind of weird at this point in my opinion